flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -150,11 +159,30 @@ public static RestClient forUrl(Configuration configuration, Executor executor,
 
     public RestClient(Configuration configuration, Executor executor)
             throws ConfigurationException {
-        this(configuration, executor, null, -1);
+        this(configuration, executor, null, -1, DefaultSelectStrategyFactory.INSTANCE);
```

Comment:
```suggestion
        this(configuration, executor, null, -1);
```
Why do you think that change is necessary? Isn't it better to have the default behavior defined in only one code location?

Reply:
Once again, just wasn't sure what Flink's preferred style is—reducing the number of constructor indirections or reducing the number of default values materialized. :)

Happy to change.

Reply:
True, I overlooked that we're doing it your way in for the default port `-1`. Feel free to revert it if you want to be more consistent in this class :innocent: